kusto/query_builder.go

```diff
@@ -506,6 +507,185 @@ func NewStmt(query stringConstant, options ...StmtOption) Stmt {
 	return s
 }
 
+// AddDatabase will add a normalized Database string to the Stmt. This is similar to the + operator on two strings,
+// except it only can be done with string constants. This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddDatabase(query string) Stmt {
+	return s.NormalizeName(query, false)
+}
+
+// AddTable will add a normalized Table string to the Stmt. This is similar to the + operator on two strings,
+// except it only can be done with string constants. This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddTable(query string) Stmt {
+	return s.NormalizeName(query, false)
+}
+
+// AddColumn will add a normalized Column string to the Stmt. This is similar to the + operator on two strings,
+// except it only can be done with string constants. This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddColumn(query string) Stmt {
+	return s.NormalizeName(query, false)
+}
+
+// AddFunction will add a normalized Function string to the Stmt. This is similar to the + operator on two strings,
+// except it only can be done with string constants. This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddFunction(query string) Stmt {
+	return s.NormalizeName(query, false)
+}
+
+// NormalizeName normalizes a string in order to be used safely in the engine.
+func (s Stmt) NormalizeName(query string, forceNormalization bool) Stmt {
```

Comment:
Tests? for everything? we don't want to mess it up

Reply:
@yogilad So should we close it or should we write tests? (and if so - how? never done that :) )

Reply:
A) PRs should include tests for new / changed code where possible.
B) here you have two valid options (and you'd probably want to cover both)
* Add unit tests which compose a query statement and validate the output string is as expected
* Add an end to end test which uses this syntax in queries and commands to be sure the output is accepted by the service (this can be an addition of scenarios to the existing query / command tests)